R/divipola.R

```diff
@@ -0,0 +1,270 @@
+#' Retrieve DIVIPOLA table
+#'
+#' @description
+#' Retrieve DIVIPOLA table including departments and municipalities
+#'
+#' @examples
+#' divipola <- divipola_table()
+#'
+#' @return \code{data.frame} object with DIVIPOLA table
+#'
+#' @export
+divipola_table <- function() {
+  dataset_path <- retrieve_support_path("DIVIPOLA")
+  divipola_table <- retrieve_table(dataset_path)
+  return(divipola_table)
+}
+
+#' Retrieve departments' DIVIPOLA codes from names
+#'
+#' @description
+#' Retrieve departments' DIVIPOLA codes from their names
+#'
+#' @param department_name character vector with the names of the departments
+#'
+#' @examples
+#' dptos <- c("TOLIMA", "HUILA", "AMAZONAS")
+#' name_to_code_dep(dptos)
+#'
+#' @return character vector with the DIVIPOLA codes of the departments
+#'
+#' @export
+name_to_code_dep <- function(department_name) {
+  checkmate::assert_character(department_name)
+
+  divipola <- divipola_table()
+  dptos <- divipola[
+    !duplicated(divipola$departamento),
+    c("codigo_departamento", "departamento")
+  ]
+  fixed_tokens <- iconv(
+    tolower(dptos$departamento),
+    from = "utf-8",
+    to = "ASCII//TRANSLIT"
+  )
+  departments_codes <- NULL
+  for (dpto in department_name) {
+    input_token <- iconv(tolower(dpto),
+      from = "UTF-8",
+      to = "ASCII//TRANSLIT"
+    )
+    distances <- as.data.frame(stringdist::afind(
+      x = fixed_tokens,
+      pattern = input_token,
+      method = "cosine",
+      window = max(8, nchar(input_token), na.rm = TRUE)
+    ))
+    min_distance_i <- which(distances$distance == min(distances$distance))
+    if (length(min_distance_i) == 1) {
+      dpto_code <- dptos$codigo_departamento[min_distance_i]
+    } else {
+      min_location <- distances$location[min_distance_i]
+      min_location_i <- min_distance_i[which.min(min_location)]
+      dpto_code <- dptos$codigo_departamento[min_location_i]
+    }
+    if (min(distances$distance) > 0.11) {
+      warning(dpto, " cannot be found as a department name")
+      departments_codes <- c(departments_codes, NA)
+    } else {
+      departments_codes <- c(departments_codes, dpto_code)
+    }
+  }
```

Comment:
> Two changes (that I'm leaving you copy to the other functions):
> 
> * functions that can be extracted out of the for loop are extracted
> * the result vector is initialized at the right length, as growing vectors in loop is very inefficient

Thanks @Bisaloo. We removed the dynamic sized vector and extracted some functions out of the for loop. `distances()` however, can't be extracted, since the window set is dynamic for each department consulted. This was included in #115.